django/db/migrations/state.py

```diff
@@ -187,14 +212,24 @@ def add_field(self, app_label, model_name, name, field, preserve_default):
         else:
             field = field
         self.models[app_label, model_name].fields[name] = field
+        if self._relations is not None:
+            model_key = (app_label, model_name)
+            concretes, _ = self._get_concrete_models_mapping_and_proxy_models()
+            real_apps = set(self.real_apps)
+            self.resolve_model_field_relations(model_key, concretes, real_apps, name, field)
         # Delay rendering of relationships if it's not a relational field.
         delay = not field.is_relation
         self.reload_model(app_label, model_name, delay=delay)
 
     def remove_field(self, app_label, model_name, name):
         model_state = self.models[app_label, model_name]
         old_field = model_state.fields.pop(name)
-        # Delay rendering of relationships if it's not a relational field.
+        if self._relations is not None:
+            model_key = (app_label, model_name)
+            concretes, _ = self._get_concrete_models_mapping_and_proxy_models()
+            real_apps = set(self.real_apps)
+            self.resolve_model_field_relations(model_key, concretes, real_apps, name, old_field)
```

Comment:
The 5 lines above look like a copy of the same 5 lines from further up. Should this be a helper method?

Reply:
These lines will be unnecessary with the new version of `resolve_model_field_relations()` (see #14781) because it  uses `self.real_apps` and handles `concretes`.